lib/widgets/search_widget.dart

```diff
@@ -1,37 +1,45 @@
 import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:flutter/gestures.dart';
 import 'package:flutter/material.dart';
-import 'package:pso_messanger/configuration/app_text.dart';
+import '../helper/user_constants.dart';
+import '../screens/conversation_screen.dart';
+import '../configuration/app_text.dart';
 import '../widgets/search_tile_widget.dart';
 import '../configuration/app_path.dart';
 import '../services/database.dart';
 import '../configuration/app_colors.dart';
 import '../widgets/appbar_widget.dart';
 
-class Search extends StatefulWidget {
+class SearchWidget extends StatefulWidget {
   @override
-  _SearchState createState() => _SearchState();
+  _SearchWidgetState createState() => _SearchWidgetState();
 }
 
-class _SearchState extends State<Search> {
-  DatabaseMethods databaseMethods = DatabaseMethods();
-  TextEditingController searchTextEdittingController = TextEditingController();
+class _SearchWidgetState extends State<SearchWidget> {
+  DatabaseMethods _databaseMethods = DatabaseMethods();
+  TextEditingController _searchTextEdittingController = TextEditingController();
 
   QuerySnapshot searchSnapshot;
 
-  initiateSearch() {
-    databaseMethods
-        .getUserByUserName(searchTextEdittingController.text)
-        .then((value) {
-      setState(() {
-        searchSnapshot = value;
+  Future<void> initiateSearch() async => await _databaseMethods
+          .getUserByUserName(_searchTextEdittingController.text)
+          .then((value) {
+        setState(() {
+          searchSnapshot = value;
+        });
       });
-    });
-  }
 
   createChatroomAndStartConversation(String userName) {
-    //List<String> users = [userName, ];
-    //databaseMethods.createChatRoom()
+    String chatRoomId = _getChatRoomId(userName, UserConstants.username);
+
+    List<String> users = [userName, UserConstants.username];
+    Map<String, dynamic> chatRoomMap = {
+      "users": users,
+      "chatroomId": chatRoomId
```

Comment:
be careful with map keys written in such way. Somethimes its better to move this to some kind of file with key_constants,  rather than raw keys